app/dcr_api/biz/pkg/storage/gcs.go

```diff
@@ -0,0 +1,117 @@
+package storage
+
+import (
+	"compress/gzip"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"github.com/pkg/errors"
+
+	credentials "cloud.google.com/go/iam/credentials/apiv1"
+	credentialspb "cloud.google.com/go/iam/credentials/apiv1/credentialspb"
+)
+
+type GoogleCloudStorage struct {
+	ctx            context.Context
+	bucket         string
+	client         *storage.Client
+	googleAccessId string
+}
+
+func NewGoogleCloudStorage(ctx context.Context, bucket string) (*GoogleCloudStorage, error) {
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create storage client")
+	}
+	serviceAccount, err := getGoogleServiceAccount()
+
+	return &GoogleCloudStorage{
+		ctx:            ctx,
+		bucket:         bucket,
+		client:         client,
+		googleAccessId: serviceAccount,
+	}, nil
+}
+
+func (g *GoogleCloudStorage) Close() {
+	g.client.Close()
+}
+
+func (g *GoogleCloudStorage) BucketPath() string {
+	return fmt.Sprintf("gs://%s", g.bucket)
+}
+
+func (g *GoogleCloudStorage) UploadFile(reader io.Reader, remotePath string, compress bool) error {
+	writer := g.client.Bucket(g.bucket).Object(remotePath).NewWriter(g.ctx)
+	defer writer.Close()
+	if compress {
+		gzipWriter := gzip.NewWriter(writer)
+		if _, err := io.Copy(gzipWriter, reader); err != nil {
+			return errors.Wrap(err, "failed to copy content to gzip writer")
+		}
+		defer gzipWriter.Close()
+	} else {
+		if _, err := io.Copy(writer, reader); err != nil {
+			return errors.Wrap(err, "failed to copy content to writer")
+		}
+	}
+	return nil
+}
+
+func (g *GoogleCloudStorage) PresignedUrl(remotePath string, method string, expires time.Duration) (string, error) {
+	if method != "GET" && method != "PUT" {
+		return "", errors.Wrap(fmt.Errorf("unkown method for signed url, supported are GET and PUT"), "")
+	}
+	c, err := credentials.NewIamCredentialsClient(g.ctx)
+	if err != nil {
+		panic(err)
```

Comment:
please return error instead of panic?

In general, try to not use `panic` in a way that some runtime error completely kills the API in the middle. The runtime error should be handled as gracefully as possible, such that it falls back to some handler, or notify error and go on. Any critical error like this may cause `panic`, but should be only at the early stage of the program (e.g., when API starts up), and not in the middle.